app/Models/Ban.php

```diff
@@ -25,62 +26,68 @@ class Ban extends Model
         'repealed_at',
     ];
 
-    public function scopeLocal(Builder $query)
+    public function scopeLocal(Builder $query): Builder
     {
         return $query->where('type', BanTypeConstants::LOCAL);
     }
 
-    public function scopeNetwork(Builder $query)
+    public function scopeNetwork(Builder $query): Builder
     {
         return $query->where('type', BanTypeConstants::NETWORK);
     }
 
-    public function scopeNotRepealed(Builder $query)
+    public function scopeNotRepealed(Builder $query): Builder
     {
         return $query->whereNull('repealed_at');
     }
 
-    public function scopeRepealed(Builder $query)
+    public function scopeRepealed(Builder $query): Builder
     {
         return $query->whereNotNull('repealed_at');
     }
 
-    public function user()
+    public function user(): BelongsTo
     {
         return $this->belongsTo(User::class);
     }
 
-    public function banner()
+    public function banner(): BelongsTo
     {
         return $this->belongsTo(User::class, 'banner_id');
     }
 
-    public function reason()
+    public function reason(): BelongsTo
     {
         return $this->belongsTo(Reason::class);
     }
 
-    public function end()
+    public function end(): bool
     {
         if (! $this->ends_at) {
             $this->ends_at = Carbon::now();
             $this->save();
+
+            return true;
         }
+
+        return false;
     }
 
-    public function repeal()
+    public function repeal(): bool
```

Comment:
Does this need to return a bool?

Reply:
Was basing on returning the result of the `->save()`. Have updated to actually do that!